packages/core/src/render3/instructions.ts

```diff
@@ -1815,23 +1806,29 @@ function invokeDirectivesHostBindings(tView: TView, viewData: LView, tNode: TNod
     const def = tView.data[i] as DirectiveDef<any>;
     const directive = viewData[i];
     if (def.hostBindings) {
-      const previousExpandoLength = expando.length;
-      setCurrentDirectiveDef(def);
-      def.hostBindings !(RenderFlags.Create, directive, tNode.index - HEADER_OFFSET);
-      setCurrentDirectiveDef(null);
-      // `hostBindings` function may or may not contain `allocHostVars` call
-      // (e.g. it may not if it only contains host listeners), so we need to check whether
-      // `expandoInstructions` has changed and if not - we still push `hostBindings` to
-      // expando block, to make sure we execute it for DI cycle
-      if (previousExpandoLength === expando.length && firstTemplatePass) {
-        expando.push(def.hostBindings);
-      }
+      invokeHostBindings(def, expando, directive, tNode, firstTemplatePass);
     } else if (firstTemplatePass) {
       expando.push(null);
     }
   }
 }
 
+export function invokeHostBindings(
```

Comment:
This name isn't very descriptive since we invoke host bindings in update mode in a very different way later. Can you change it?

```suggestion
export function invokeHostBindingsInCreationMode(
```